test/python/circuit/test_library.py

```diff
@@ -230,6 +230,52 @@ def test_non_symmetric_raises(self, adjacency_matrix):
             GraphState(adjacency_matrix)
 
 
+@ddt
+class TestFourierCheckingLibrary(QiskitTestCase):
+    """Test the Fourier Checking circuit."""
+
+    def assertFourierCheckingIsCorrect(self, f_truth_table, g_truth_table, fc_circuit):
+        """Assert that the Fourier Checking circuit produces the correct matrix."""
+
+        simulated = Operator(fc_circuit)
+
+        num_qubits = int(np.log2(len(f_truth_table)))
+
+        # create Hadamard matrix, re-use approach from TestMCMT
+        h_i = 1 / np.sqrt(2) * np.array([[1, 1], [1, -1]])
+        h_tot = np.array([1])
+        for _ in range(num_qubits):
+            h_tot = np.kron(h_tot, h_i)
+
+        f_mat = np.diag(f_truth_table)
+        g_mat = np.diag(g_truth_table)
+
+        expected = np.linalg.multi_dot([h_tot, g_mat, h_tot, f_mat, h_tot])
+        expected = Operator(expected)
+
+        self.assertTrue(expected.equiv(simulated))
+
+    @data(
+        ([1, -1, -1, -1], [1, 1, -1, -1])
+    )
+    @unpack
+    def test_fourier_checking(self, f_truth_table, g_truth_table):
```

Comment:
If there is only one case, why `data`?
Either something like this
```suggestion
    def test_fourier_checking(self, f_truth_table=[1, -1, -1, -1], g_truth_table=[1, 1, -1, -1]):
```
or something like this
```suggestion
    @data(([1, -1, -1, -1], [1, 1, -1, -1]),
          ([1, 1, 1, 1], [1, 1, 1, 1]))
    @unpack
    def test_fourier_checking(self, f_truth_table, g_truth_table):
```

Reply:
That is a good point here, thanks for that! I went for the second approach, since I assumed having one more test wouldn't hurt. I have executed the "Commit suggestion".